### DISCUSSION TITLE
How does one back up Koreader data?

### DISCUSSION THREAD

**Discussion by drvenom123** [koreader/koreader#10338] [Q&A]
I'm starting to use koreader and I'm really digging it, at this point I'm obsessed, specially the book map option. I highlight and take notes quite a bit. This is cool because it shows up on the book map and one gets to see note taking patterns across the book. The issue is that I'll probably upgrade my ereader (kobo libra 2) whenever there's a new one. How do I carry (save) all this highlights and notes data tied to epubs, so that I can basically put the epub in the new ereader and have koreader still see the highlights and notes where they were on the previous reader, and, have the book map option still capture all that information? Is that even possible? Or once you move to a new device, the epub and highlights get decoupled and you'd have to go back to each book and highlight everything and add your notes manually?

**Comment by Frenzie** [comment]
> When you open a document in KOReader, it creates a directory alongside the document with same name as the document. This directory contains a .LUA file which includes all the highlights and user options (formatting properties etc.) related to that document. You can manually backup/restore these directories.

But if you're moving from a Kobo device to another Kobo device, you can probably copy literally everything.
